src/bulletin/bulletin.service.ts

```diff
@@ -91,4 +91,54 @@ export class BulletinService {
       }
     });
   }
+
+  async deleteBulletin(id: number) {
+    try {
+      const dateNow = new Date();
+      const bulletin = await this.prismaService.bulletin.findUnique({
+        where: { id },
+      });
+
+      if (!bulletin) {
+        throw new HttpException(
+          `Bulletin with id ${id} does not exist`,
+          HttpStatus.NOT_FOUND,
+        );
+      }
+
+      if (bulletin.deleted_at) {
+        throw new HttpException(
+          `Bulletin with id ${id} has already been deleted`,
+          HttpStatus.BAD_REQUEST,
+        );
+      }
+
```

Comment:
Use a database transaction to ensure that both the bulletin deletion and PDF attachment upload (if applicable) are completed successfully. If any part of the process fails, the transaction should be rolled back. Refer to this url: https://www.prisma.io/docs/orm/prisma-client/queries/transactions